pkg/cloud/uris_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cloud_test
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+	// Imported to register a known URL
+	_ "github.com/cockroachdb/cockroach/pkg/cloud/amazon"
```

Comment:
I'd have a slight preference to just test this in pkg/cloud/amazon over making this test in the common lib depend on the implementation that in turn depends on the lib, but since it is only a test this isn't a hill I'll die on if you like it here.

Reply:
I was being lazy, I think what this test should probably do is register an external storage schema with some custom variables for redaction and just test that.